src/mysql_query.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+#
+# igcollect - PostgreSQL query results
+#
+# This script executes a single query and prints the results.  The columns
+# returned by the query are going to be appended to the given prefix.
+# The query must return numeric values.
+#
+# Copyright (c) 2016, InnoGames GmbH
+#
+from __future__ import print_function
+
+import MySQLdb
+
+from argparse import ArgumentParser
+from time import time
+
+
+def parse_args():
+    parser = ArgumentParser()
+    parser.add_argument('--prefix', default='mysql_query')
+    parser.add_argument('--dbname', default='mysql')
+    parser.add_argument(
+        '--query',
+        required=True,
+        action='append',
+        dest='queries', )
+    parser.add_argument('--key-column')
+    return parser.parse_args()
+
+
+def main():
+    """The main program"""
+    args = parse_args()
+    items = []
+    now = str(int(time()))
+
+    db = MySQLdb.connect(
+        user='root',
+        host="localhost",
+        db=args.dbname,
+        read_default_file='/etc/mysql/my.cnf', )
+    cur = db.cursor(MySQLdb.cursors.DictCursor)
+    for query in args.queries:
+        cur.execute(query)
+        if not cur.rowcount:
+            raise Exception('No result')
+        else:
+            rows = cur.fetchall()
+            items.extend(
+                get_row_data(args.key_column, rows) if args.key_column
+                else get_column_data(rows))
+
+    for key, value in items:
+        print(args.prefix + '.' + key, value, now)
+
+
+def get_column_data(rows):
+    if len(rows) > 1:
+        raise Exception('Multiple rows')
+    for row in rows:
+        return row.items()
+
+
+def get_row_data(key_column, rows):
```

Comment:
I would put `rows` as the first argument to make the function signature compatible with the one above.